### PR TITLE
Better iso + jak 2 + cleanup

### PR DESCRIPTION
**Pull request by Zedb0T**
Seems to work end to end.

### CODE REVIEW

File: buildlaunchercoreEXE.bat [hunk 1]
```diff
@@ -1,8 +1,8 @@
 
 set mypath=%~dp0
-pyinstaller --onefile openGOALModLauncher.py --icon resources\appicon.ico 
+pyinstaller --onefile openGOALModLauncher.py --icon resources\appicon.ico --collect-data "pycdlib" --hidden-import=pycdlib datas=[(path.join(site_packages,"pycdlib")],
```
Comment (dallmeyer) [inline buildlaunchercoreEXE.bat:3]: remove

File: openGOALModLauncher.py [hunk 2]
```diff
@@ -203,11 +223,13 @@ def exitWithError():
 INCLUDE_UNINSTALLED = True
 LATEST_TABLE_SORT = [6, False] # wakeup sorted by last launch date
 
-
 def getRefreshedTableData(sort_col_idx):
     # uncomment/comment the next two lines if you want to test with a local file
-    mod_dict = requests.get("https://raw.githubusercontent.com/OpenGOAL-Unofficial-Mods/OpenGoal-ModLauncher-dev/main/resources/jak1_mods.json").json()
-    # mod_dict = json.loads(open("resources/jak1_mods.json", "r").read())
+    # mod_dict = requests.get("https://raw.githubusercontent.com/OpenGOAL-Unofficial-Mods/OpenGoal-ModLauncher-dev/main/resources/jak1_mods.json").json()
+    mod_dict = json.loads(open("resources/jak1_mods.json", "r").read())
```
Comment (dallmeyer) [inline openGOALModLauncher.py:229]:
```suggestion
    mod_dict = requests.get("https://raw.githubusercontent.com/OpenGOAL-Unofficial-Mods/OpenGoal-ModLauncher-dev/main/resources/jak1_mods.json").json()
    # mod_dict = json.loads(open("resources/jak1_mods.json", "r").read())
```

File: openGOALModLauncher.py [hunk 3]
```diff
@@ -275,9 +299,9 @@ def getRefreshedTableData(sort_col_idx):
                         (mod["videos_url"] if "videos_url" in mod else ""),
                         (mod["photos_url"] if "photos_url" in mod else ""),
                         (mod["image_override_url"] if "image_override_url" in mod else ""),
+                        (mod["game"] if "game" in mod else "")
```
Comment (dallmeyer) [inline openGOALModLauncher.py:302]: else jak1? not sure it matters yet

File: utils/launcherUtils.py [hunk 4]
```diff
@@ -160,30 +191,35 @@ def reinstall(MOD_ID):
...
+            if pathlib.Path(iso_path).is_file:
+                if not (pathlib.Path(iso_path).suffix).lower() == ".iso" :
+                    print((pathlib.Path(iso_path).suffix).lower())
+                    1 / 0
+    print("Extraction completed successfully.")
     # Close Gk and goalc if they were open.
     try_kill_process("gk.exe")
     try_kill_process("goalc.exe")
     print("Done update starting extractor\n")
+    print(currentOS + " POOOOOOOOOOOOOOOP")
```
Comment (dallmeyer) [inline utils/launcherUtils.py:215]: lol

File: utils/launcherUtils.py [hunk 5]
```diff
@@ -439,25 +576,116 @@ def launch(URL, MOD_ID, MOD_NAME, LINK_TYPE):
...
+        if(NotExtracted):
+            print("Error! Iso data does not appear to be extracted to " + UniversalIsoPath +"//" + GAME + "//" + "Z6TAIL.DUP")
+            print("Will ask user to provide ISO")
+        if(NotCompiled):
+            print("Error! The game is not compiled")
+        if((LastWrite < LatestRel)):
+            print("Looks like we need to download a new update!")
+            print(LastWrite)
+            print(LatestRel)
+            print("Is newest posted update older than what we have installed? " + str((LastWrite < LatestRel)))
```
Comment (dallmeyer) [inline utils/launcherUtils.py:682]: should it error out here? and not launch

**Review by dallmeyer — CHANGES_REQUESTED** [review]

File: utils/launcherUtils.py (hunk 4 above)
Comment (Zedb0T) [reply utils/launcherUtils.py:215]: IMG_3328_9000

File: openGOALModLauncher.py (hunk 3 above)
Comment (Zedb0T) [reply openGOALModLauncher.py:302]: I dont think so

File: utils/launcherUtils.py (hunk 5 above)
Comment (Zedb0T) [reply utils/launcherUtils.py:682]: This should never get hit, but I figured checking a second time didn't hurt

**Pull request merged**